packages/core/admin/ee/server/utils/persisted-tables.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+/**
+ * Finds all tables in the database that start with a prefix
+ * @param {string} prefix
+ * @returns {Array}
+ */
+const findTablesThatStartWithPrefix = async (prefix) => {
+  const tables = await strapi.db.dialect.schemaInspector.getTables();
+  return tables.filter((tableName) => tableName.startsWith(prefix));
+};
+
+/**
+ * Get all reserved table names from the core store
+ * @returns {Array}
+ */
+const getPersistedTables = async () =>
+  strapi.store.get({
+    type: 'core',
+    key: 'persisted_tables',
+  }) ?? [];
```

Comment:
I think the ?? [] won't work here as the store.get promise has to be awaited first

Reply:
ah yes you're right